File: 백준/Silver/1522. 문자열 교환/문자열 교환.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on('line', (line) => {
  input.push(line);
}).on('close', () => {
  let word = [...input.pop()];
  const Word = word.length;
  const first = word[0];
  let window = 0;
  for (let i = 0; i < Word; i++) {
    if (word[i] === first) window += 1;
  }

  word = word.concat(word);
  let answer = Word;
  for (let i = 0; i < Word; i++) {
    let now = 0;
    for (let j = 0; j < window; j++) {
      if (word[i + j] !== first) now += 1;
    }
    answer = Math.min(answer, now);
  }
  console.log(answer);
  process.exit();
});

File: 백준/Silver/1522. 문자열 교환/문자열 교환.py
import sys
input=sys.stdin.readline

l=list(input().rstrip())
cnt=0
start=-1
last=-1
loc=[]
for i in range(len(l)):
  if l[i]=='b':
    loc.append(i)
    cnt+=1
    if start==-1:
      start=i
    last=i

if cnt==0:
  print(0)
  exit()

l+=l
answer=-1
for item in loc:
  now=0
  for i in range(item,item+cnt):
    if l[i]!='b':
      now+=1
  if answer==-1 or answer>now:
    answer=now
    if answer==0:
      break

print(answer)
